fiona/ogrext.pyx

```diff
@@ -1227,11 +1227,13 @@ cdef class WritingSession(Session):
             if record["geometry"] is None:
                 return True
             return record["geometry"]["type"].lstrip("3D ") in valid_geom_types
-
-        log.debug("Starting transaction (initial)")
-        result = gdal_start_transaction(self.cogr_ds, 0)
-        if result == OGRERR_FAILURE:
-            raise TransactionError("Failed to start transaction")
+        transactions_supported = check_capability_transaction(self.cogr_ds)
+        log.debug("Transaction supported: {}".format(transactions_supported))
+        if transactions_supported:
```

Comment:
Great! Alternative would be to move the test into `gdal_start_transaction` itself but we could do that layer.